login/pam.c

```diff
@@ -164,6 +164,22 @@ int login_prompt(glome_login_config_t *config, pam_handle_t *pamh,
   if (strlen(token) >= input_size) {
     return failure(EXITCODE_PANIC, error_tag, "pam-authtok-size");
   }
+
+  // OpenSSH provides fake password when login is not allowed,
+  // for example due to PermitRootLogin set to 'no'
+  // https://github.com/openssh/openssh-portable/commit/283b97
+  const char fake_password[] =
+      "\b\n\r\177INCORRECT";  // auth-pam.c from OpenSSH
+  bool is_fake = true;
+
+  // Constant-time comparison in case token contains user's password
```

Comment:
I guess there's no advantage of moving the constant-time comparison function generically into crypto.c, right? (Alongside is_zeroed.)

Reply:
Note that this comparison function is "special" as it loops around fake_password so that it matches the lenght of token's lenght. I don't have a strong preference either way.